index.bs

```diff
@@ -184,12 +179,15 @@ Integration with HTML {#html}
 
 This specification integrates with the [[!HTML]] specification by patching the algorithms below:
 
-Document object initialization {#document-init}
+Navigation fetch {#navigation-fetch}
+----------
+
+At [=process a navigate fetch=], after step 13.4 call [$retrieve the Client Hints set$] with <var ignore>currentURL's origin</var> as input.
```

Comment:
Here you are retrieving the CH set, but not doing anything with it. Also, it's not clear to me that you need to do anything special here to have navigation requests have client hints.
Step 13.5 in "process a navigate fetch" is calling [fetch](https://fetch.spec.whatwg.org/#concept-fetch), which then calls [append CH to request](https://wicg.github.io/client-hints-infrastructure/#abstract-opdef-append-client-hints-to-request)

Reply:
We weren't clearly loading the client hints set for the navigation that was being fetched, I think this could be moved into that step though. https://github.com/WICG/client-hints-infrastructure/issues/42

Reply:
OK, so in this case I think that what you want to do here is retrieve the CH set, and store it as the settingObject's CH set. Maybe worthwhile to generalize the first 2 steps of update the CH set and call that from both places?